packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -495,10 +526,10 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                 ).apply {
                                     dispatchReceiver = irGetObject(propertyClass.companionObject()!!.symbol)
                                     var arg = 0
-                                    // Name
-                                    putValueArgument(arg++, irString(entry.key))
+                                    // Persisted name
+                                    putValueArgument(arg++, irString(persistedName))
                                     // Public name
-                                    putValueArgument(arg++, irString(""))
+                                    putValueArgument(arg++, irString(publicName))
```

Comment:
Wouldn't [this test](https://github.com/realm/realm-kotlin/blob/27308439fd02e57d2e08a976b7c401d19a5d816b/packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/PersistedNameTests.kt#L319) cover that scenario?

Or do you mean specifically test for when (a) the public name is `""` then (b) is set to a non-empty string? (Rather than (a) non-empty string then (b) a different non-empty string as in the existing test). In this case, I think we can just add a similar test as the one I linked to but have the `oldSchema` be a class with a property without the `@PersistedName` (since all public names will be `""` if not using the annotation), and have the `newSchema` be a class with a property with a `@PersistedName`.

Reply:
Ahhh. I thought that `publicName` was set to the same as `persistedName` when there was no `@PersistedName` on it. I guess we don't change anything in what is fed to the C-API then. 

I needed at some point we could verify that we don't trigger a migration by adding a pre-change realm-file and verify that we don't trigger migrations, but I guess that would require #577

Reply:
Okay, yeah no the public name is always `""` unless `@PersistedName` is used. Yeah when #577 is done we could add another test using a realm file.